sdks/java/extensions/zetasketch/src/main/java/org/apache/beam/sdk/extensions/zetasketch/HllCount.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.zetasketch;
+
+import com.google.zetasketch.HyperLogLogPlusPlus;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * {@code PTransform}s to compute HyperLogLogPlusPlus (HLL++) sketches on data streams based on the
+ * <a href="https://github.com/google/zetasketch">ZetaSketch</a> implementation.
+ *
+ * <p>HLL++ is an algorithm implemented by Google that estimates the count of distinct elements in a
+ * data stream. HLL++ requires significantly less memory than the linear memory needed for exact
+ * computation, at the cost of a small error. Cardinalities of arbitrary breakdowns can be computed
+ * using the HLL++ sketch. See this <a
+ * href="http://static.googleusercontent.com/media/research.google.com/en/us/pubs/archive/40671.pdf">published
+ * paper</a> for details about the algorithm.
+ *
+ * <p>HLL++ functions are also supported in <a
+ * href="https://cloud.google.com/bigquery/docs/reference/standard-sql/hll_functions">Google Cloud
+ * BigQuery</a>. Using the {@code HllCount PTransform}s makes the interoperation with BigQuery
+ * easier.
```

Comment:
Done. (Previously said "easier" because users could also depend directly on ZetaSketch and implement their own pipeline logic to handle sketches themselves, instead of using `HllCount`, but I agree the description now makes more sense.)